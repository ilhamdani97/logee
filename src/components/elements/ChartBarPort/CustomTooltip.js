import React from 'react';
import style from './styles.scoped.css';
import PropTypes from 'prop-types';
import { thousand } from '../../../utils/format';


const CustomTooltip = (props) => {
  const{ active, payload,customProps } = props;


  const convertToCurrency = (isCurrency,val) => {
    let currency = isCurrency ? `Rp ${thousand(val)}`: val;

    return currency;
  };
  if (active && payload && payload.length) {

    const captionUpper = payload[0].payload[customProps.titleUpper.value];
    const captionBottom = payload[0].payload[customProps.titleBottom.value];
    const bgUpper = [style[customProps.titleUpper.tooltipColor],style.tooltipRadiusUpper].join(' ');
    const bgBottom = [style[customProps.titleBottom.tooltipColor],style.tooltipRadiusBottom].join(' ');
    const subCapUp = customProps.titleUpper.subCaption;
    const subCapBtm = customProps.titleBottom.subCaption;
    return (
      <div className={style.tooltip}>
        <div className={bgUpper}>
          {`
          ${subCapUp.textPosition === 'left' ? subCapUp.textValue : ''}
          ${convertToCurrency(customProps.titleUpper.currency, captionUpper)} 
          ${subCapUp.textPosition === 'right' ? subCapUp.textValue : ''}
          `}

        </div>
        <div className={bgBottom}>
          {`
          ${subCapBtm.textPosition === 'left' ? subCapBtm.textValue : ''}
          ${convertToCurrency(customProps.titleBottom.currency,  captionBottom)} 
          ${subCapBtm.textPosition === 'right' ? subCapBtm.textValue : ''}`}
        </div>

      </div>
    );
  }

  return null;
};


CustomTooltip.defaultProps = {
  active:false,
  customProps:{},
  payload: [],
};

CustomTooltip.propTypes = {
  active:PropTypes.bool,
  customProps: PropTypes.object,
  payload: PropTypes.array,
};


export default CustomTooltip;
