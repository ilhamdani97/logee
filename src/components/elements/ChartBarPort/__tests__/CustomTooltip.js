import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CustomTooltip from '../CustomTooltip';

const customProps =  {
  'titleUpper': {
    'currency': false,
    'value': 'totalOrderCount',
    'subCaption': {
      'textPosition': 'right',
      'textValue': 'Pesanan'
    },
    'tooltipColor': 'primary'
  },
  'titleBottom': {
    'currency': true,
    'value': 'totalTransaction',
    'subCaption': {
      'textPosition': 'right',
      'textValue': ''
    },
    'tooltipColor': 'primary'
  }
};

const payload = [{ 'fill':'#487A9D','dataKey':'yAxisBar','name':'yAxisBar','color':'#487A9D','value':0,'payload':{ 'totalTransactionExport':0,'totalTransactionImport':0,'totalTransactionAll':0,'periodNodeId':11,'periodNode':'Nov','yAxisUnit':0,'xAxisBar':0,'yAxisBar':0,'yStacked':0 } },{ 'fill':'#e8590c','dataKey':'xAxisBar','name':'xAxisBar','color':'#e8590c','value':0,'payload':{ 'totalTransactionExport':0,'totalTransactionImport':0,'totalTransactionAll':0,'periodNodeId':11,'periodNode':'Nov','yAxisUnit':0,'xAxisBar':0,'yAxisBar':0,'yStacked':0 } }];

describe('src/components/elements/CustomTooltip', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<CustomTooltip
      active={true}
      customProps={customProps}
      payload={payload}/>);
    expect(tree).toMatchSnapshot();
  });
});
