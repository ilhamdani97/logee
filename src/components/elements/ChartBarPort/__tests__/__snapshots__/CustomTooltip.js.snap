// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/CustomTooltip render 1`] = `
<div>
  <div
    className=" "
  >
    
          
          undefined 
          Pesanan
          
  </div>
  <div
    className=" "
  >
    
          
          Rp NaN 
          
  </div>
</div>
`;
